mmv1/third_party/cai2hcl/common/converter_map.go

```diff
@@ -0,0 +1,44 @@
+package common
+
+import (
+	"github.com/GoogleCloudPlatform/terraform-google-conversion/v5/caiasset"
+)
+
+// Configuration object to map CAI Assets to Converters.
+type ConverterMap struct {
+	// Mapping from CAI Asset Type to converter name.
+	AssetTypeToConverter map[string]string
+
+	// Mapping from converter name and converter.
+	Converters map[string]Converter
+}
+
+// ConvertMap implements Converter interface to be able to convert any Asset type.
+func (c ConverterMap) Convert(assets []*caiasset.Asset) ([]*HCLResourceBlock, error) {
```

Comment:
If I understand this correctly, it means
grouping used to be 
- 'google_project' group: project1, project2, billing1
- 'google_network' group: network1, network2

And your comment mentioned this should be changed to grouping like this?  probably need to mix the name + type together as the hash key. 
- project1, billing1
- project2
- network1
- network2

I think it makes sense, and the difference is the same type of converter will be initiated multiple times rather than 1 time in the current style of grouping. 

---
`tpg_provioder.Provider()` can be stored as a global variable if the schema is required at multiple invocation of converter init.

---

Merging like the tf->cai way is not that easy in this context. mainly because cty.Value is not that easy to manipulate like json.

Reply:
It does make sense. It all sounds legit.

One thing I don't yet know is how to implement getName(asset Asset) { ... } function in each converter, when we think of it from generator perspective.

I.e. I planned to expose AssetType and SelfLinkRegExp like [this](https://github.com/amirkaromashkin/magic-modules/blob/main-cai2hcl/mmv1/templates/cai2hcl/resource_converter.go.erb#L43). 

I will continue looking how to do that.

Reply:
As discussed offline with Stephen - there is no need to group assets by asset name due to current converters architecture.

PTAL - I have just significantly simplified the overall architecture: 
1. Got rid of multi-layered converter maps => all converters are installed in a single file converter_map
2. Got rid of UberConverters => one converter_map is left.
2. Got rid of converter factories
3. Moved converter names constants inside the Converters (not sure if this is a good choice)

I tried to simplify it even further and converter converter_map to converters array ([here](https://github.com/amirkaromashkin/terraform-google-conversion/commit/c8833078c01087e333bd36cef5b23ac4a03bb21b#diff-f25bf0a8d306a73f506a5e38af63497535b086d6e1a79d201fb1091e5a2007bb)), but I believe we current approach is actually a better one (clearly states which converters affect which asset types)

One thing I'm not sure is wether it is really needed to move Converter Name inside the converters: current converter names are exactly TF schemas, but its not necessarily so. So leaving them as raw strings, I think was a fine choice either.